metaserver/src/main/scala/scala/meta/languageserver/LSPLogger.scala

```diff
@@ -14,10 +14,13 @@ class LSPLogger(@BeanProperty var encoder: PatternLayoutEncoder)
   import LSPLogger._
 
   override def append(event: ILoggingEvent): Unit = {
-    val message =
-      if (encoder != null) new String(encoder.encode(event), UTF_8)
-      else event.getFormattedMessage
-    connection.foreach(_.logMessage(MessageType.Log, message))
+    // Skip rpc message noise.
+    if (!event.getLoggerName.startsWith("langserver.core.Message")) {
```

Comment:
this can be configured moreflexibly in `logback.groovy`. I'll push this change directly on the PR.